gapis/api/gles/dependency_graph_behaviour_provider.go

```diff
@@ -215,6 +215,22 @@ func (*GlesDependencyGraphBehaviourProvider) GetBehaviourForAtom(
 			case *GlTexSubImage2D:
 				texData, _ := getTextureDataAndSize(ctx, cmd, id, s, c.Bound.TextureUnit, cmd.Target, cmd.Level)
 				b.Modify(g, texData)
+			case *GlGenerateMipmap:
+				tex, err := subGetBoundTextureOrErrorInvalidEnum(ctx, cmd, id, nil, s, GetState(s), cmd.Thread(), nil, cmd.Target)
+				if err != nil {
+					log.E(ctx, "Can not find bound texture %v", cmd.Target)
+				}
+				for levelIndex, level := range tex.Levels.Range() {
+					for layerIndex := range level.Layers.Range() {
+						data, size := tex.dataAndSize(levelIndex, layerIndex)
+						if levelIndex == 0 {
+							b.Read(g, data)
+						} else {
+							b.Read(g, size)
+							b.Write(g, data)
```

Comment:
Write both data and size

Reply:
Does it write to size?

Reply:
Writing to size breaks tests and actual texture reads.

Reply:
I would say it writes to size, yes.  The spec is not 100% clear though.

"Mipmap generation replaces texture image levels levelbase + 1 through q with
images derived from the levelbase image, regardless of their previous contents."

My interpretation is that the previous non-zero levels are discarded (not-read), and completely replace (defined data+size).  In particular, my reading is that it is valid for the previous definitions of the images to be of wrong size, or absent at all.  Consequently, DCE should be killing any definitions of non-zero levels.